@here/harp-mapview/lib/Tile.ts

```diff
@@ -447,6 +445,9 @@ export class Tile implements CachedResource {
 
     private m_resourceInfo: TileResourceInfo | undefined;
 
+    // List of owned textures for disposal
+    private m_ownedTextures: Set<string> = new Set();
```

Comment:
This can be `Set<THREE.Texture>`.
IMHO, it's more straighforard and probably a little bit  efficient as we search will compare by reference.

Reply:
I wanted to avoid holding a reference to the texture, but since we own the texture it should be fine to use the texture directly.

Reply:
>  wanted to avoid holding a reference to the texture,

For this case you have WeakMap/Set [1] in your toolbox. But i agree, in this case there is no special reason to use WeakSet, normal set is enough.

[1]https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet

Reply:
Ah, good to know. I think I will use WeakSet b/c it makes more clear that textures are not held in this set but in the material.
Maybe for some weird reason somebody disposes the material and the texture from outside it would be even safer.

Reply:
Good point ... 
+2